lib/data/dataset.js

```diff
@@ -82,6 +82,14 @@ const getDataset = (xml) =>
     else if (!semverSatisfies(version.get(), '2022.1.0 - 2024.1.x'))
       throw Problem.user.invalidEntityForm({ reason: `Entities specification version [${version.get()}] is not supported.` });
 
+    let warnings;
+    if (semverSatisfies(version.get(), '<2024.1.x'))
+      warnings = {
```

Comment:
I think it's nice to avoid a `let` where possible. How about `const` + a ternary?

```js
const warnings = semverSatisfies(version.get(), '>=2024.1.x')
  ? null
  : {
    type: 'oldEntityVersion',
    // ...
  };
```